### DISCUSSION TITLE
Run Music Assistant as an addon

### DISCUSSION THREAD

**Discussion by gieljnssns** [music-assistant/support#971] [Feature requests and ideas]
If we can run Music Assistant as an addon, then the music keeps on playing when Home-Assistant restarts..

**Comment by marcelveldt** [comment]
Well, this is an interesting one...

When I started this project 2 years ago I created it as an add-on.
The add-on only supported google cast, Sonos and slimproto.

The positive was that the server/stream engine was complete detached from HA core, just like a standalone Logitech Media Server, Z-Wave JS etc. Music streaming was not dependent of HA core. In fact the whole MA frontend could be run as standalone app.

This approach had 3 big downsides:

1) For player support I needed to implement every player in Music Assistant which felt strange as HA already has good integrations to talk to each player. 

2) A separate add-on required to run the server, adding more complexity.

3) Overhead, because I also needed to handle an API layer with authentication etc.


So at some point I decided (after some feedback talk with various people) to simplify things and convert the whole project to just a HA integration with the big benefit of 2 things:

1) Direct access to all players supported by HA
2) A lot of boilerplate already there such as API framework and authentication/security.

This comes with a big downside however and that is that now the music streaming is attached to HA core process.
HA core down = Music server down. And another side effect is that HA core could potentially have negative impact on streaming when there are performance issues and vice versa. 

So, this feature request is interesting to see if there is a real interest to decouple the stream engine again from HA core.
Maybe work out a hybrid form where the most popular players are provided by MA natively and the rest can be provided by HA.
It will also solve the issue of duplicate players because MA itself is the player provider.

**Reply by gieljnssns** [reply]
Thanks for your answer.

**Comment by gieljnssns** [comment]
https://github.com/music-assistant/hass-music-assistant/issues/979#issuecomment-1301829788

**Comment by gieljnssns** [comment]
I think [this](https://github.com/music-assistant/hass-music-assistant/issues/997) issue is also gone when using MA as an addon

**Comment by Selim042** [comment]
This could also be interesting if the add-on could use ingress to allow outside access to the player.

**Comment by marcelveldt** [comment]
Just as an FYI the V2 release of MA which I'm working on contains this feature request because the only way I could solve some of the challenges I had, were to detach the MA process from HA.

And yes, that also means it will be possible to run the frontend standalone

**Comment by OzGav** [comment]
Old V1 discussion. Closing.
